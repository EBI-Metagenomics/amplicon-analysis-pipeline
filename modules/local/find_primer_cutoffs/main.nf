
process FIND_PRIMER_CUTOFFS {
    // Find inflection points in conservation curves
    tag "$meta.id"
    label 'very_light'
    container "oras://community.wave.seqera.io/library/mi-pimento:0.0.4--1795182b3e13ee89"

    input:
    tuple val(meta), path(bcv)

    output:
    tuple val(meta), path("*cutoffs.tsv"), emit: cutoffs
    path "versions.yml"                  , emit: versions

    script:
    """
    if [[ -s ./$bcv ]]; then
        pimento find_cutoffs -i ${bcv} -o ${meta.id}_${meta.var_region}
    else
        touch ${meta.id}_${meta.var_region}_cutoffs.tsv
    fi

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        mi-pimento: \$( pimento --version | cut -d" " -f3 )
    END_VERSIONS
    """
}
