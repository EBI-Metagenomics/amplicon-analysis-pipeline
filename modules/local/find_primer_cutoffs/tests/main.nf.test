nextflow_process {

    name "Test Process FIND_PRIMER_CUTOFFS"
    script "../main.nf"
    process "FIND_PRIMER_CUTOFFS"
    tag "modules"
    tag "find_primer_cutoffs"

    test("Should run without failures") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false, var_region:'16S-V3-V4' ], // meta map
                    file('${projectDir}/modules/local/find_primer_cutoffs/tests/data/test_mcp_cons.tsv', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.cutoffs).match("cutoffs") }
            )
        }

    }

}
