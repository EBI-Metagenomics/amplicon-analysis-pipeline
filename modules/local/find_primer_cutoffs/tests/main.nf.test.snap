{
    "inf_points_out": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "16S-V3-V4",
                    "test_16S-V3-V4_inf_points.tsv:md5,b4da973b056b9b5ed4ec7ca8d7998be4"
                ]
            ]
        ],
        "timestamp": "2023-11-03T15:59:01.72783491"
    }
}